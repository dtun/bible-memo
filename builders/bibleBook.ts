import { build, perBuild } from "@jackfranklin/test-data-bot";
import { faker } from "@faker-js/faker";
import { BibleBook } from "@/types";

export let bibleBookBuilder = build<BibleBook>({
  fields: {
    id: perBuild(faker.string.uuid),
    name: perBuild(faker.lorem.word),
    nameLong: perBuild(faker.lorem.word),
    abbreviation: perBuild(faker.lorem.word),
    bibleId: perBuild(faker.string.uuid),
  },
});

export let bibleBooks = {
  50: [...new Array(50)].map(bibleBookBuilder),
};
