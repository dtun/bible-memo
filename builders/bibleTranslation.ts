import { build, perBuild, oneOf } from "@jackfranklin/test-data-bot";
import { faker } from "@faker-js/faker";
import { BibleTranslation } from "@/types";

export let bibleTranslationBuilder = build<BibleTranslation>({
  fields: {
    id: perBuild(faker.string.uuid),
    dblId: perBuild(faker.string.uuid),
    relatedDbl: null,
    name: perBuild(faker.lorem.word),
    nameLocal: perBuild(faker.lorem.word),
    abbreviation: perBuild(faker.lorem.word),
    abbreviationLocal: perBuild(faker.lorem.word),
    description: perBuild(faker.lorem.sentence),
    descriptionLocal: perBuild(faker.lorem.sentence),
    language: {
      id: perBuild(faker.string.uuid),
      name: perBuild(faker.lorem.word),
      nameLocal: perBuild(faker.lorem.word),
      script: perBuild(faker.lorem.word),
      scriptDirection: oneOf("LTR", "RTL"),
    },
    countries: [
      {
        id: perBuild(faker.string.uuid),
        name: perBuild(faker.lorem.word),
        nameLocal: perBuild(faker.lorem.word),
      },
    ],
    type: oneOf("text", "audio", "video"),
    updatedAt: perBuild(() => faker.date.recent().toString()),
    audioBibles: [],
  },
});

export let bibleTranslations = {
  50: [...new Array(50)].map(bibleTranslationBuilder),
};
