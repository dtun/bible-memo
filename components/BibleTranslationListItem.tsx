import { Pressable, StyleSheet } from "react-native";

import { Text, View } from "@/components/Themed";
import { type BibleTranslation } from "@/types";

function BibleTranslationListItem({
  item: bible,
}: {
  item: BibleTranslation & {
    selected?: boolean;
    onPress: (id: string) => void;
  };
}) {
  return (
    <Pressable
      accessibilityLabel={bible.nameLocal}
      accessibilityRole="link"
      accessibilityState={{ selected: bible.selected }}
      hitSlop={24}
      onPress={() => bible.onPress(bible.id)}
      style={({ pressed }) => ({
        opacity: pressed ? 0.5 : 1,
      })}
    >
      <View
        darkColor="transparent"
        lightColor="transparent"
        style={styles.listItem}
      >
        <Text style={styles.title}>{bible.abbreviationLocal}</Text>
        {bible.selected ? <Text style={styles.check}>âœ“</Text> : null}
      </View>
      <Text style={styles.body}>{bible.nameLocal}</Text>
    </Pressable>
  );
}

export { BibleTranslationListItem };

let styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  listEmpty: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
    alignSelf: "center",
  },
  body: {
    fontSize: 16,
    marginHorizontal: 4,
  },
  title: {
    fontSize: 20,
    fontWeight: "bold",
    marginVertical: 4,
    marginHorizontal: 4,
  },
  check: {
    fontSize: 24,
    fontWeight: "bold",
    marginVertical: 4,
    marginHorizontal: 8,
  },
  separator: {
    marginVertical: 16,
    height: 1,
    width: "100%",
  },
  listItem: {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-between",
  },
});
