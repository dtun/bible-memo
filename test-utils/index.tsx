import { render } from "@testing-library/react-native";

const AllTheProviders = ({ children }: { children: React.ReactElement }) => {
  return <>{children}</>;
};

const customRender = (ui: React.ReactElement) =>
  render(ui, { wrapper: AllTheProviders });

// re-export everything
export * from "@testing-library/react-native";

// override render method
export { customRender as render };
