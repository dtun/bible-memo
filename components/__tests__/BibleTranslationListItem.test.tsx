import * as React from "react";

import { render, screen, userEvent } from "@/test-utils";
import { bibleTranslationBuilder } from "@/builders/bibleTranslation";
import { BibleTranslationListItem } from "../BibleTranslationListItem";

let data = bibleTranslationBuilder();
let onPress = jest.fn();

it(`renders <BibleTranslationListItem />`, () => {
  render(<BibleTranslationListItem item={{ ...data, onPress }} />);

  expect(screen).toMatchSnapshot();
});

it(`renders selected state`, () => {
  render(
    <BibleTranslationListItem item={{ ...data, onPress, selected: true }} />
  );

  expect(screen.getByLabelText(data.nameLocal)).toHaveProperty(
    "props.accessibilityState",
    { selected: true }
  );

  screen.rerender(
    <BibleTranslationListItem item={{ ...data, onPress, selected: false }} />
  );

  expect(screen.getByLabelText(data.nameLocal)).toHaveProperty(
    "props.accessibilityState",
    { selected: false }
  );
});

it(`calls onPress with id`, async () => {
  render(<BibleTranslationListItem item={{ ...data, onPress }} />);

  await userEvent.press(screen.getByLabelText(data.nameLocal));

  expect(onPress).toHaveBeenCalledWith(data.id);
});
