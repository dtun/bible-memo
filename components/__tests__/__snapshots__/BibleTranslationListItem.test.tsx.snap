// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <BibleTranslationListItem /> 1`] = `
<View
  accessibilityLabel="vado"
  accessibilityRole="link"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  hitSlop={24}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "transparent",
        },
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#000",
          },
          {
            "fontSize": 20,
            "fontWeight": "bold",
            "marginHorizontal": 4,
            "marginVertical": 4,
          },
        ]
      }
    >
      absorbeo
    </Text>
  </View>
  <Text
    style={
      [
        {
          "color": "#000",
        },
        {
          "fontSize": 16,
          "marginHorizontal": 4,
        },
      ]
    }
  >
    vado
  </Text>
</View>
`;
